### DISCUSSION TITLE
How to get feature importance through gain?

### DISCUSSION THREAD

**Discussion by hanhanwu** [microsoft/FLAML#482] [Q&A]
Hello FLAML team,

I tried getting feature importance through FLAML like what's said here: https://microsoft.github.io/FLAML/docs/FAQ/

But when using model like LGBM, the importance is using the count of splits. I want to check the feature importance through the "gain". Is that possible to get it through FLAML?

I have also tried to set `save_feature_importance_type=1` as what's shown here: https://lightgbm.readthedocs.io/en/latest/Parameters.html#saved_feature_importance_type, but when calling FLAML's estimator's feature_importance_, it's still using split counts....

Any suggestion? Thank you!

**Comment by qingyun-wu** [comment]
If LGBM supports it, you should be able to also get it through FLAML's LGBM if configured properly. 
1. What is the LGBM version you are using? As far as I can see, the `save_feature_importance_type` is only effective after v3.3.0. 
2. If you are using LBGM with a version >= v3.3.0.  Can you show me your code on how you set`save_feature_importance_type=1` for LGBM while using FLAML?

**Reply by hanhanwu** [reply]
My LGBM is v3.3.2, here's the code of FLAML settings.
![Screen Shot 2022-03-15 at 3 17 20 PM](https://user-images.githubusercontent.com/4024769/158873282-767d7bf6-827a-46fc-b6a5-49c39ddd244b.png)
Then I was calling the feature importance through FLAML trained model like what's shown here: https://microsoft.github.io/FLAML/docs/FAQ/, the importance was still ranked by splits not by gain

**Reply by qingyun-wu** [reply]
@[hanhanwu](https://github.com/hanhanwu), can you successfully use gain-based feature importance even if you manually configure an LGBMClassifier? According to the documentation you provided, the parameter `can be used only in CLI version`.

Thanks!

**Reply by hanhanwu** [reply]
I see. Makes sense, thank you @qingyun-wu
